tests/unit/src/test/scala/tests/BatchedFunctionSuite.scala

```diff
@@ -95,11 +99,51 @@ class BatchedFunctionSuite extends BaseSuite {
 
     mkString.unpause()
 
-    assertDiffEqual(paused.value, None)
-    assertDiffEqual(paused2.value, None)
-
-    val unpaused2 = mkString(List("a", "b"))
-    assertDiffEqual(unpaused2.value, Some(Success("ab")))
+    for {
+      _ <- paused.failed
+      _ <- paused2.failed
+      res <- mkString(List("a", "b"))
+      _ = assertEquals(res, "ab")
+    } yield ()
+  }
 
+  test("cancel2") {
+    val executorService = Executors.newFixedThreadPool(10)
+    val ec2 = ExecutionContext.fromExecutor(executorService)
+    var i = 1
+    val stuckExample: BatchedFunction[String, String] =
+      new BatchedFunction(
+        (seq: Seq[String]) => {
+          seq.toList match {
+            case "loop" :: Nil =>
+              val future = Future.apply {
```

Comment:
I don't think this will ever be cancelled, we should add a flag in the while that changes on cancelation. Scala futures are not cancelable

Reply:
`var I` (line 113) is the flag

Reply:
Ach ok, not sure why I haven't noticed before :sweat_smile: